### DISCUSSION TITLE
unable to view all the labels from metrics in grafana

### DISCUSSION THREAD

**Discussion by mdvaliyullah** [grafana/mimir#3412] [Help and support]
unable to view all the labels from metrics in grafana. After re-starting mimir it shows all the labels from the same metrics.

- Running mimir in monolithic mode on docker with nginx as load balancer and 3 mimir containers
- Configured same volume for all the 3 mimir containers.
- Observed below in the logs its for each refresh in grafana the request is going to one of mimir containers.

_play-grafana-mimir-mimir-1-1_ | level=info ts=2022-11-08T11:05:42.460052587Z caller=handler.go:225 user=fake msg="query stats" component=query-frontend method=POST path=/prometheus/api/v1/query_range response_time=4.966792ms query_wall_time_seconds=0.002526542 **fetched_series_count=5** fetched_chunk_bytes=631 **fetched_chunks_count=5** sharded_queries=0 split_queries=1 param_step=60 param_end=1667905500 param_query="metric_name{Host=\"abc\"}" param_start=1667905200

_play-grafana-mimir-mimir-2-1_ | level=info ts=2022-11-08T11:05:57.376894052Z caller=handler.go:225 user=fake msg="query stats" component=query-frontend method=POST path=/prometheus/api/v1/query_range response_time=5.398875ms query_wall_time_seconds=0.003711333 **fetched_series_count=6** fetched_chunk_bytes=756 **fetched_chunks_count=6** sharded_queries=0 split_queries=1 param_step=60 param_end=1667905500 param_query="metric_name{Host=\"abc\"}" param_start=1667905200

- I tried enabling sharding but it didn't worked. Please advise how to fix this issue

**Comment by Logiraptor** [comment]
> unable to view all the labels from metrics in grafana. After re-starting mimir it shows all the labels from the same metrics.

Can you give an example of the output you see and what you expected?

> Configured same volume for all the 3 mimir containers.

Mimir is not designed to share a directory like this, can you try giving each container a separate volume and report if anything changes?
